java/core/src/main/java/com/microsoft/bond/BondType.java

```diff
@@ -12,38 +12,52 @@
 import java.util.concurrent.ConcurrentHashMap;
 
 /**
- * Represents a type in the Bond type system and implements behavior specific to that type, such as
- * schema inference, initialization, serialization and deserialization. A Bond type cane be a primitive,
- * a struct, a specialization of a generic type (including containers, nullable, or generic Bond structs).
+ * A type descriptor in the Bond type system that implements behavior specific to a particular type, such
+ * as schema inference, initialization, serialization and deserialization. A Bond type can be a primitive,
+ * an enum, a struct, or a specialization of a generic type (including containers, nullable, or generic
+ * Bond structs). Each Bond type is associated with a single {@link Class} instance of the value, although
+ * multiple Bond types can use the same value class (e.g. specializations of a generic type, or using the
+ * same Java type to represent multiple primitive types in Bond such as signed vs. unsigned integers).
  * @param <T> the class of the value
  */
 public abstract class BondType<T> {
 
-    // The global type cache
-    static final TypeCache typeCache = new TypeCache();
-
-    // Registry of struct type resolvers; entries are added by static class initializers
-    private static final ConcurrentHashMap<
-            Class<? extends BondSerializable>,
-            StructBondTypeResolver<? extends BondSerializable>> structTypeResolverRegistry =
-            new ConcurrentHashMap<
-                    Class<? extends BondSerializable>,
-                    StructBondTypeResolver<? extends BondSerializable>>();
-
-    // Registry of enum types; entries are added by static class initializers
-    private static final ConcurrentHashMap<
-            Class<? extends BondEnum>,
-            EnumBondType<? extends BondEnum>> enumTypeRegistry =
-            new ConcurrentHashMap<
-                    Class<? extends BondEnum>,
-                    EnumBondType<? extends BondEnum>>();
-
-    // precomputed hash code helps to speed up object retrieval from the type cache
-    private final int precomputedHashCode;
-
-    // restrict subclasses to the current package
-    BondType(int precomputedHashCode) {
-        this.precomputedHashCode = precomputedHashCode;
+    // The global type cache, which maps a Bond type descriptor to itself and is used to cache type descriptor
```

Comment:
I don't understand your comment. The purpose of this data structure is to retrieve a value given a key, which is what Map offers. Set API doesn't have that.

---
In reply to: [124083933](https://github.com/Microsoft/bond/pull/502#discussion_r124083933) [](ancestors = 124083933)